src/vs/workbench/services/views/browser/viewDescriptorService.ts

```diff
@@ -348,6 +361,7 @@ export class ViewDescriptorService extends Disposable implements IViewDescriptor
 
 		this.viewContainersCustomLocations.clear();
 		this.viewDescriptorsCustomLocations.clear();
+		this.viewContainerBadgeEnablementStates.clear();
```

Comment:
This method might be called by "Reset View Locations" which is a little strange. Either the method should take parameters or we need new functions.

Reply:
It is, but I wonder if this is a good place to put it for reset view state. If not I can move it out and we can just create a separate reset badge

Reply:
I think it should be resettable, but we need a separate command as people look for reset view locations